pkg/streamingpromql/operator/instant_vector_selector.go

```diff
@@ -89,7 +85,23 @@ func (v *InstantVectorSelector) NextSeries(_ context.Context) (InstantVectorSeri
 			continue
 		}
 
-		data.Floats = append(data.Floats, promql.FPoint{T: stepT, F: val})
+		// if (val, h have been set by PeekPrev, we do not know if val is 0 because that's the actual value, or because
```

Comment:
[nit] Looks like a typo

```suggestion
		// if (val, h) have been set by PeekPrev, we do not know if val is 0 because that's the actual value, or because
```